broadcaster/backlog/backlog.go

```diff
@@ -54,6 +56,26 @@ func (b *backlog) Head() BacklogSegment {
 	return b.head.Load()
 }
 
+func (b *backlog) backlogSizeInBytes() uint64 {
+	if b.head.Load() == nil || b.tail.Load() == nil {
+		return 0
+	}
+	headSeg := b.head.Load()
+	headSeg.messagesLock.RLock()
+	headMsg := headSeg.messages[0]
+	headSeg.messagesLock.RUnlock()
+
+	tailSeg := b.tail.Load()
+	tailSeg.messagesLock.RLock()
+	tailMsg := tailSeg.messages[len(tailSeg.messages)-1]
```

Comment:
Let's gracefully handle headSeg.messages or tailSeg.messages being empty. I don't think this should be possible but we've had similar panics in tests and it's better safe than sorry. Ideally we'd return an error here which we'd log as a warning and then just not update the gauge.

Reply:
Makes sense, I'll add checks to handle that case